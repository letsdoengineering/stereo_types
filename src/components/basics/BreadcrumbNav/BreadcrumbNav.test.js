import Breadcrumb from './BreadcrumbNav'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

describe('BreadcrumbNav', () => {
    const urlList = [
        { label: 'home', url: '/' },
        { label: 'page1', url: '/page1/' },
        { label: 'current location' },
    ]
    it('has not changed the JSX layout of the page from previous snapshot', () => {
        const tree = renderer.create(<Breadcrumb urlList={urlList} />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('renders the page DOM that is accessible', async () => {
        const { container } = render(<Breadcrumb urlList={urlList} />)
        const results = await axe(container)

        expect(results).toHaveNoViolations()
    })
})
