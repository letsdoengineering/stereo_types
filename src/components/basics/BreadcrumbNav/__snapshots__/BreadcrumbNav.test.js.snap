// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbNav has not changed the JSX layout of the page from previous snapshot 1`] = `
<ul
  className="list listUnstyled"
>
  <li
    className=""
  >
    <a
      className="link"
      href="/"
    >
      <span
        className="sizeS weightNormal textGray"
      >
        home
      </span>
    </a>
  </li>
  <li
    className=""
  >
    <span
      className="sizeS weightNormal textGray divider"
    >
      &gt;
    </span>
    <a
      className="link"
      href="/page1/"
    >
      <span
        className="sizeS weightNormal textGray"
      >
        page1
      </span>
    </a>
  </li>
  <li
    className=""
  >
    <span
      className="sizeS weightNormal textGray divider"
    >
      &gt;
    </span>
    <span
      className="sizeS weightBold textGray"
    >
      current location
    </span>
  </li>
</ul>
`;
